base/sparse/linalg.jl

```diff
@@ -527,18 +527,13 @@ function normestinv{T}(A::SparseMatrixCSC{T}, t::Integer = min(2,maximum(size(A)
     X = Array(Ti, n, t)
     X[1:n,1] = 1
     for j = 2:t
-        repeated = true
-        while repeated
+        while true
             for i = 1:n
-                X[i,j] = rand()>=0.5?1:-1
+                X[i,j] = rand(-1:2:1)
```

Comment:
Wow ... it would never have occurred to me to even measure the performance of this one line.

I can make it significantly faster by not going into the loop and just calling (a 10X speed up over `rand(-1:2:1)`, but still 3X slower than the @mfasi's original code), but I suppose this still suffers from the allocation of temporary variables @jiahao pointed out.